### PR TITLE
Added Overlay Hero styles and a bit of cleanup

### PR DESCRIPTION
**Pull request by jimmynotjim**
Update for pages that need an overlay hero that uses a BG image instead of the inset right image.

## Additions

- Added `overlay` and `white-text` modifiers to the hero

## Changes

- Updated the Hero styles to properly match the content layout and
  simplified the rules

## Testing

_See [ CF Testing Components Locally for more info](https://github.com/cfpb/capital-framework/blob/canary/CONTRIBUTING.md#testing-components-locally)_

- run `npm run build` in the `[ cf repo ]`
- cd to `[ cf repo ]/tmp/cf-layout` and run `npm link`
- Used the linked module in your project by cd-ing to your project root and running `npm link cf-layout`
- Build your project
- Add `hero__overlay` to your hero's classlist
- Add an inline bg image style to the `hero_wrapper` Ex. `style="background-image: url(http://beta.consumerfinance.gov/static/img/home-hero_lg.jpg);`

## Review

- @schaferjh
- @ajbush 
- @KimberlyMunoz 
- @sebworks 
- @anselmbradford 

## Screenshots

screen shot 2016-03-24 at 3 26 30 pm

screen shot 2016-03-24 at 3 24 23 pm

## Notes

- This is using the Homepage image even though it's not the homepage. It's what we have on beta currently that makes testing across machines simple. The flag/building image will need to be set in the CMS when the new page is published.
- After discussion with @ajbush, we decided it was better to keep the image the same as the default hero at smaller screen sizes, rather than span edge-to-edge and remove the bottom padding. It's more consistent and leads to a simpler code base.

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices 
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

**Comment by Scotchester** [comment]
I have no clue why the PR build is failing, but I have a guess about why the Push build is failing: It's still running on npm < 3. Should we update the Node version in `_travis.yml` to specify Node 5+? @contolini

**Comment by Scotchester** [comment]
All good now; @jimmynotjim, can you take care of the merge conflicts?

**Comment by jimmynotjim** [comment]
Bueller...?

bueller

File: src/cf-layout/src/cf-layout.less [hunk 1]
```diff
@@ -650,36 +650,117 @@
...
+       allowing you to easily override them before compiling."
+  patterns:
+  - name: Sizes
+    codenotes:
+      - |
+        @hero-desktop-height
+        @hero-image-height
+  tags:
+  - less
+*/
```
Comment (Scotchester) [inline src/cf-layout/src/cf-layout.less]: We're not using Topdoc comments anymore.

Comment (jimmynotjim) [inline src/cf-layout/src/cf-layout.less]: Ah dang, forgot about that, got overly zealous updating cfgov-refresh comments.

File: src/cf-layout/src/cf-layout.less [hunk 2]
```diff
@@ -650,36 +650,35 @@
     });
   }
 }
-//
-// Hero
-//
```
Comment (Scotchester) [inline src/cf-layout/src/cf-layout.less]: We still want this comment header in there, though.

**Comment by Scotchester** [comment]
Nothing jumps out at me in the code, other than wondering if this falls back adequately for the lack of `:not` support in IE8. (I know we were already using one of them, that I think I introduced, and probably didn't test thoroughly, but it occurred to me to wonder this time...)

I will try to test this afternoon.

File: src/cf-layout/src/cf-layout.less [hunk 3]
```diff
@@ -650,36 +650,35 @@
...
 .hero {
-    background-color: @block__bg;
+    background-color: @hero-bg;
 
-    &_wrapper {
-        box-sizing: border-box;
-        padding: unit(@grid_gutter-width / @base-font-size-px, em)
-                 unit(15px / @base-font-size-px, em);
+    .respond-to-min( @bp-sm-min, {
+        display: table;
```
Comment (Scotchester) [inline src/cf-layout/src/cf-layout.less]: From a quick test on tax-time-saving, moving `display: table;` to `.hero` breaks the `vertical-align: middle;` on `.hero_text` and `.hero_image`.

**Comment by schaferjh** [comment]
Just poking my head in here because we have an unfinished hero waiting on this over in V1... @anselmbradford @sebworks @KimberlyMunoz any review you guys can provide in Jimmy's absence?

File: src/cf-layout/src/cf-layout.less (hunk 3 above)
Comment (KimberlyMunoz) [inline src/cf-layout/src/cf-layout.less]: Does L746 take care of this?

File: src/cf-layout/src/cf-layout.less [hunk 4]
```diff
@@ -650,36 +650,35 @@
...
-        .respond-to-min(@bp-xl-min, {
-            padding-right: 0;
-            padding-left: 0;
+    &_wrapper {
+        padding-top: unit( @grid_gutter-width / @base-font-size-px, em );
+        padding-bottom: unit( @grid_gutter-width / @base-font-size-px, em );
+
+        .respond-to-min( @bp-sm-min, {
+            padding-right: 15px;
+            padding-left: 15px;
```
Comment (anselmbradford) [inline src/cf-layout/src/cf-layout.less]: Should these have unit conversion?

File: src/cf-layout/src/cf-layout.less (hunk 3 above)
Comment (Scotchester) [inline src/cf-layout/src/cf-layout.less]: L746 is necessary, but not sufficient. The things that are `display: table-cell;` apparently need their immediate parent to be `display: table;`.

**Comment by anselmbradford** [comment]
LGTM :+1:

File: src/cf-layout/src/cf-layout.less (hunk 3 above)
Comment (jimmynotjim) [inline src/cf-layout/src/cf-layout.less]: I'm not seeing this problem locally. Are you sure that spacing under the text isn't making it slightly shorter than the image? I see that here:

screen shot 2016-04-04 at 3 49 17 pm

But it's not due to `table` and `table-cell`

Comment (Scotchester) [inline src/cf-layout/src/cf-layout.less]: Here's what I'm seeing:

screen shot 2016-04-04 at 17 23 47

After putting `display: table` back on `.hero_wrapper`:

screen shot 2016-04-04 at 17 25 52

**Comment by jimmynotjim** [comment]
Updated with a height correction after debugging with @Scotchester

**Comment by Scotchester** [comment]
:+1:

**Pull request merged**